Mathlib/Analysis/Calculus/FDeriv/Pow.lean

```diff
@@ -0,0 +1,221 @@
+/-
+Copyright (c) 2025 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Analysis.Calculus.FDeriv.Mul
+import Mathlib.Analysis.Calculus.FDeriv.Comp
+
+/-!
+# Derivative of `(f x) ^ n`, `n : ℕ`
+
+In this file we prove that `(x ^ n)' = n * x ^ (n - 1)`, where `n` is a natural number.
+Additionally, we prove the case for non-commutative rings,
+where the result is instead `∑ i ∈ Finset.range n, f x ^ (n.pred - i) •> f' <• f x ^ i`.
+
+For a more detailed overview of one-dimensional derivatives in mathlib, see the module docstring of
+`Analysis/Calculus/Deriv/Basic`.
+
+## Keywords
+
+derivative, power
+-/
+
+variable {𝕜 𝔸 E : Type*}
+
+section NormedRing
+variable [NontriviallyNormedField 𝕜] [NormedRing 𝔸] [NormedAddCommGroup E]
+variable [NormedAlgebra 𝕜 𝔸] [NormedSpace 𝕜 E] {f : E → 𝔸} {f' : E →L[𝕜] 𝔸} {x : E} {s : Set E}
+
+open scoped RightActions
+
+private theorem aux (f : E → 𝔸) (f' : E →L[𝕜] 𝔸) (x : E) (n : ℕ) :
+    f x •> ∑ i ∈ Finset.range (n + 1), f x ^ ((n + 1).pred - i) •> f' <• f x ^ i
+      + f' <• (f x ^ (n + 1)) =
+    ∑ i ∈ Finset.range (n + 1 + 1), f x ^ ((n + 1 + 1).pred - i) •> f' <• f x ^ i := by
+  rw [Finset.sum_range_succ _ (n + 1), Finset.smul_sum]
+  simp only [Nat.pred_eq_sub_one, add_tsub_cancel_right, tsub_self, pow_zero, one_smul]
+  simp_rw [smul_comm (_ : 𝔸) (_ : 𝔸ᵐᵒᵖ), smul_smul, ← pow_succ']
+  congr! 5 with x hx
+  simp [Nat.lt_succ_iff] at hx
+  rw [tsub_add_eq_add_tsub hx]
+
+theorem HasStrictFDerivAt.pow' (h : HasStrictFDerivAt f f' x) (n : ℕ) :
```

Comment:
```suggestion
theorem HasStrictFDerivAt.fun_pow' (h : HasStrictFDerivAt f f' x) (n : ℕ) :
```
Could you do this consistently throughout the PR (this is mostly already the case) and add the `.pow'` versions when missing?

Reply:
Done I think, could you check if I missed any?